src/identity/identity-wallet.ts

```diff
@@ -403,9 +448,14 @@ export class IdentityWallet implements IIdentityWallet {
     private readonly _credentialWallet: ICredentialWallet,
     private readonly _opts?: {
       credentialStatusPublisherRegistry?: CredentialStatusPublisherRegistry;
+      prover?: IZKProver;
     }
   ) {
     this._credentialStatusPublisherRegistry = this.getCredentialStatusPublisherRegistry(_opts);
+    if (_opts?.prover) {
+      this._prover = _opts?.prover;
```

Comment:
Then we have to pass the `_circuitStorage` in the constructor to be able to create the NativeProver
```
    this._prover = opts?.prover ?? new NativeProver(_circuitStorage);
```
We should pass then the prover with `opts.prover` or pass the `_circuitStorage` and if none of the 2 options are passed throw an error.

Reply:
not sure if we need default one, prover now is optional (since I can use wallet only for Eth identities). I would leave it like it is. If we go with `_circuitStorage` approach I need to pass this even if I don't need it (in case of Eth identities).

Reply:
Yes, to prevent breaking changes I see more natural to pass `prover` in `IdentityWallet.transitState` and then removing `_prover` property from `IdentityWallet`. If you agree I'll make the updates with this approach.